### DISCUSSION TITLE
Is merging with scheduled workouts even possible?

### DISCUSSION THREAD

**Discussion by jameskyle** [philosowaffle/peloton-to-garmin#667] [General]
First, this project should get some kind of reward. Seriously useful.

I've been doing the garmin coach adaptive workout plans. Like most, I turn on the watch, do the workout on the peloton, delete the watch workout, sync the peloton workout using p2g.

I've noticed this messes with the coach scheduling a good bit. It still uses your stress scores, but it counts the scheduled workouts skipped and replans the week. 

What I'm wondering is if it's even a possible feature to merge the stats being synced into an existing workout?

The flow would be the same, except you don't delete the watch workout. And when you sync, that workouts stats are updated.

I know it's not currently supported, but is there anything in the API or file spec that would prevent it?

**Comment by jameskyle** [comment]
Looked a bit at the garmin developer site and thought a bit...

It's possible to edit activities in the connect app. So it must be possible to update generally. Using the app, I can modify avg and max power.

Though this is different than updating the entire power time series, it does imply a means of updating in place in general.

**Comment by philosowaffle** [comment]
This should be technically possible, but is definitely no small feat.  In this case, there's a chance it wouldn't even solve the problem.  

From what I can tell the primary problem is Garmin Connect is not honoring the P2G workout as counting towards the scheduled workout.  This in and of itself would need more investigation to understand "what would need to be in the workout file for Garmin to recognize it and attribute it to the scheduled workout?"

My fear, is this attribution or linking to a scheduled workout will be data stored in a proprietary field P2G can't easily replicate.  Just a hunch, I could be totally wrong.

There are a couple tools online that are capable of merging FIT files the way you are proposing.   [This one](https://gotoes.org/strava/Combine_GPX_TCX_FIT_Files.php) immediately comes to mind.  While not automated, this would prove out if just a "blind merging" of files is enough for the P2G workout to be counted towards the scheduled workout or if more is needed.
